ruby/ext/google/protobuf_c/extconf.rb

```diff
@@ -2,7 +2,9 @@
 
 require 'mkmf'
 
-$CFLAGS += " -O3 -std=c99 -Wno-unused-function -DNDEBUG "
+$CFLAGS += " -O3 -std=c99 -Wno-unused-function " +
+           "-Wno-declaration-after-statement -Wno-unused-variable " +
```

Comment:
Yes, here's an example in a Travis build:

https://travis-ci.org/google/protobuf/jobs/62482683#L3477